core/src/main/scala/io/github/gaelrenoux/tranzactio/ConnectionSource.scala

```diff
@@ -39,6 +46,16 @@ object ConnectionSource {
       }
     }
 
+    def runTransactionStream[R, E, A](task: Connection => ZStream[R, E, A], commitOnFailure: => Boolean = false)
+       (implicit errorStrategies: ErrorStrategiesRef, trace: Trace): ZStream[R, Either[DbException, E], A] = {
+
+      ZStream.acquireReleaseWith(openConnection.tap(c => setAutoCommit(c, autoCommit = false)).mapError(Left(_)))
+        { c => commitConnection(c).tapEither(_ => closeConnection(c)).orDie }
+        .flatMap(c => task(c).mapError(Right(_))
+          .tapError(_ => (if (commitOnFailure) commitConnection(c) else rollbackConnection(c)).mapError(Left(_))))
```

Comment:
I'm not sure why you use a different approach over the one in `runTransaction` here?

As far as I can see, it behaves the same in case of a success (the transaction is committed). However, on a failure, the transaction is first rollbacked (line 55), then commited (line 53). It shouldn't have any adverse effect semantically, but it can impact performance.

=> Unless there's an advantage I'm missing, I prefer my approach: commit on success, rollback on failure (using `tapBoth` instead of `tapError`), and close the connection in the release step.

Reply:
Nevermind, I think I get it: the `tap` argument is run for every entry in the stream, not a the end of the stream, duh.

I'm playing a bit with it, and I think I actually like having two levels of `acquireReleaseWith`: one to get the connections, handling only opening and closing the connection, and a second one to either commit or rollback depending on what happened.